src/mappings/mappingHandlers.ts

```diff
@@ -1,23 +1,79 @@
-import { SubstrateBlock, SubstrateExtrinsic, SubstrateEvent } from '@subql/types'
-import { BlockHandler } from '../handlers/block'
-import { EventHandler, ExtrinsicHandler } from '../handlers'
-import { CallHandler } from '../handlers/call'
-import { VestingScheduleHandler } from '../handlers/vestingschedule'
+import { EventRecord } from "@polkadot/types/interfaces";
+import { SubstrateExtrinsic, SubstrateBlock } from "@subql/types";
+import { SpecVersion, Event, Extrinsic } from "../types";
 
+let specVersion: SpecVersion;
 export async function handleBlock(block: SubstrateBlock): Promise<void> {
-    const handler = new BlockHandler(block)
+  // Initialise Spec Version
+  if (!specVersion) {
+    specVersion = await SpecVersion.get(block.specVersion.toString());
+  }
 
-    await handler.save()
-}
+  // Check for updates to Spec Version
+  if (!specVersion || specVersion.id !== block.specVersion.toString()) {
+    specVersion = new SpecVersion(block.specVersion.toString());
+    specVersion.blockHeight = block.block.header.number.toBigInt();
+    await specVersion.save();
+  }
+
+  // Process all events in block
+  const events = block.events
+    .filter(
+      (evt) =>
+        !(evt.event.section === "system" &&
+        evt.event.method === "ExtrinsicSuccess")
+    )
+    .map((evt, idx) =>
+      handleEvent(block.block.header.number.toString(), idx, evt)
+    );
 
-export async function handleEvent(event: SubstrateEvent): Promise<void> {
-    const handler = new EventHandler(event)
+  // Process all calls in block
+  const calls = wrapExtrinsics(block).map((ext, idx) =>
+    handleCall(`${block.block.header.number.toString()}-${idx}`, ext)
+  );
 
-    await handler.save()
+  // Save all data
+  await Promise.all([
+    store.bulkCreate("Event", events),
+    store.bulkCreate("Extrinsic", calls),
+  ]);
 }
 
-export async function handleCall(extrinsic: SubstrateExtrinsic): Promise<void> {
-    const extrinsicHandler = new ExtrinsicHandler(extrinsic);
+function handleEvent(
+  blockNumber: string,
+  eventIdx: number,
+  event: EventRecord
+): Event {
+  const newEvent = new Event(`${blockNumber}-${eventIdx}`);
+  newEvent.blockHeight = BigInt(blockNumber);
+  newEvent.module = event.event.section;
+  newEvent.event = event.event.method;
+  return newEvent;
+}
+
+function handleCall(idx: string, extrinsic: SubstrateExtrinsic): Extrinsic {
+  const newExtrinsic = new Extrinsic(idx);
+  newExtrinsic.txHash = extrinsic.extrinsic.hash.toString();
+  newExtrinsic.module = extrinsic.extrinsic.method.section;
+  newExtrinsic.call = extrinsic.extrinsic.method.method;
+  newExtrinsic.blockHeight = extrinsic.block.block.header.number.toBigInt();
+  newExtrinsic.success = extrinsic.success;
+  newExtrinsic.isSigned = extrinsic.extrinsic.isSigned;
+  return newExtrinsic;
+}
 
-    await extrinsicHandler.save()
+function wrapExtrinsics(wrappedBlock: SubstrateBlock): SubstrateExtrinsic[] {
```

Comment:
Suggest a better naming here
```suggestion
function extractExtrinsics(block: SubstrateBlock): SubstrateExtrinsic[] {
```